recipes/tensorstore/build.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+if [[ $target_platform =~ osx.* ]]; then
+    CLANG="$CC_FOR_BUILD" source gen-bazel-toolchain
+    # SDK's curl gets in the way
+    if [[ -d "$CONDA_BUILD_SYSROOT/usr/include/curl" ]]; then
+        mv "$CONDA_BUILD_SYSROOT/usr/include/curl" "$CONDA_BUILD_SYSROOT/usr/include/curl.do-not-use"
+    fi
+else
+    source gen-bazel-toolchain
+fi
+
+system_libs="com_google_boringssl"
+system_libs+=",org_sourceware_bzip2"
+system_libs+=",org_blosc_cblosc"
+system_libs+=",se_curl"
+system_libs+=",jpeg"
+system_libs+=",png"
+system_libs+=",org_lz4"
+system_libs+=",org_tukaani_xz"
+system_libs+=",net_zlib"
+system_libs+=",com_github_pybind_pybind11"
+system_libs+=",com_github_nlohmann_json"
+system_libs+=",org_aomedia_avif"
+# system_libs+=",com_google_absl"
```

Comment:
I guess unvendoring abseil is out of the question for now? They seem to be using [2023-03-03](https://github.com/google/tensorstore/blob/6db79b4acf150b047096509b0c307bde23b8aaa9/third_party/com_google_absl/workspace.bzl#L29), but we have [built different abseils in conda-forge](https://anaconda.org/conda-forge/libabseil/files?version=). Might not be relevant either given that it all ends up in a single SO?

Reply:
It's okay for now. I'd still like to get tensorstore to use the conda absl, but we can do it in the feedstock too.

Reply:
What are the build issues with conda absl? Updating absl version to to 2020 08 shouldn't be a big deal.

Reply:
There is no system_libs option for absl nor protobuf in upstream tensorstore. We need that for the build to pick up absl from conda.

Reply:
Ok, yeah. I haven't really looked at system libs for absl.  

protobuf is tricky, though, because of the changes that we need to build grpc and protos in cmake.